core/src/main/java/org/elasticsearch/index/query/MatchQueryParser.java

```diff
@@ -45,11 +62,9 @@ public MatchQueryBuilder fromXContent(QueryParseContext parseContext) throws IOE
         XContentParser parser = parseContext.parser();
 
         MatchQuery.Type type = MatchQuery.Type.BOOLEAN;
-        if ("match_phrase".equals(parser.currentName()) || "matchPhrase".equals(parser.currentName()) ||
-                "text_phrase".equals(parser.currentName()) || "textPhrase".equals(parser.currentName())) {
+        if (parseContext.parseFieldMatcher().match(parser.currentName(), MATCH_PHRASE_FIELD)) {
             type = MatchQuery.Type.PHRASE;
-        } else if ("match_phrase_prefix".equals(parser.currentName()) || "matchPhrasePrefix".equals(parser.currentName()) ||
-                "text_phrase_prefix".equals(parser.currentName()) || "textPhrasePrefix".equals(parser.currentName())) {
+        } else if (parseContext.parseFieldMatcher().match(parser.currentName(), MATCH_PHRASE_FIELD)) {
```

Comment:
Can you have a quick look again if this should be MATCH_PHRASE_PREFIX_FIELD instead? In which case it would also be good to catch this in tests if it is wrong.